src/algebra/group/pi.lean

```diff
@@ -235,6 +236,20 @@ lemma pi.single_mul [Π i, mul_zero_class $ f i] (i : I) (x y : f i) :
   single i (x * y) = single i x * single i y :=
 (mul_hom.single f i).map_mul x y
 
+/-- The injection into a pi group at different indices commutes.
+
+For injections of commmuting elements at the same index, see `commute.map`
+-/
+@[to_additive]
+lemma pi.mul_single_commute [Π i, mul_one_class $ f i] :
+  pairwise (λ i j, ∀ (x : f i) (y : f j), commute (mul_single i x) (mul_single j y)) :=
+begin
+  intros i j hij x y, ext k,
+  by_cases h1 : i = k, { subst h1, simp [hij], },
+  by_cases h2 : j = k, { subst h2, simp [hij], },
+  simp [h1,  h2],
+end
+
```

Comment:
This feels like it might be a good lemma too, since you prove this below as part of a lemma statement:
```suggestion

/-- The injection into a pi group with the same values commutes. -/
@[to_additive "The injection into a pi additive group with the same values commutes."]
lemma pi.mul_single_apply_commute [Π i, mul_one_class $ f i] (x : Π i, f i) (i j : I) :
  commute (mul_single i (x i)) (mul_single j (x j)) :=
begin
  obtain rfl | hij := decidable.eq_or_ne i j,
  { refl },
  { exact pi.mul_single_commute _ _ hij _ _, },
end

```

Reply:
Good idea